### DISCUSSION TITLE
Installing on External Drive

### DISCUSSION THREAD

**Discussion by abhicmraja** [ichi4096/vivado-on-silicon-mac#59] [Q&A] [answered]
Hi. I use an M1 Mac, and I don't have enough storage space on my internal SSD to install Vivado. So I'm looking to get this installed on my external drive. I have setup Docker to use the external drive, but I realised that the .sh files in the repo still use the /home/user directory by default for the installation, and hence me getting the /home/user/Xilinx does not have enough space error even after allocating 100+GB to Docker on my external drive. 

Is changing the "/home/user" to "/Volumes/<drive-name>" in all the .sh files the only way to go about this? Or is there any easier way to go about doing this? Would appreciate any help regarding this.

Thanks a lot in advance!

**Comment by ichi4096** [comment]
/home/user is a folder only visible inside the docker container. It doesn't have access to any files on the host system except to the folder You downloaded this repository to. This is because /home/user is mounted to that folder in one of the docker commands in the scripts.

I had no trouble installing onto an external SSD. What You can try to do when the setup asking You to log in:

- Don't type in Your credentials
- Open the Docker Desktop dashboard
- Select the running container
- Open the terminal (I don't know what the button or tab is called exactly)
- Type in "df -h"

This should tell You how much space the container thinks it has.

**Reply by abhicmraja** [reply]
Ohh ok.
It says home/user has 56GB, and 174GB is used...? I don't have anything on my external drive, except for the repo (only 5GB is used in total of 750GB). The docker engine/builder and Docker itself has been allocated 300GB. How do I go about it now?

This is what I'm getting for the container's space:
<img width="542" alt="Screenshot 2024-10-08 at 6 30 35 PM" src="https://github.com/user-attachments/assets/ad62b898-6495-42cf-a473-7ced011b3e80">

**Reply by ichi4096** [reply]
What's Your external drive's file system?

Could You maybe screenshot the output of `diskutil list` in your mac's terminal while your external drive is plugged in?

**Reply by abhicmraja** [reply]
<img width="609" alt="diskutil_ss" src="https://github.com/user-attachments/assets/e49d7dcb-282b-46f7-ae6f-3170264785e0">

The file system of the partition that has Docker is APFS. 

In the setup.sh, Docker is mounting /home/user/ from the internal drive (cuz the used storage on my internal drive is almost 174GB), hence the usage of 174GB on the container? Should I make a new volume on my external drive and mount that in setup.sh instead of /home/user?

**Reply by ichi4096** [reply]
I think I may see where the confusion lies. The Docker app itself doesn't need to be installed on the external drive. The `vivado-on-silicon-mac` folder  as shown by macOS is the one bound to `/home/user` in the Linux container. This folder needs to be on the external drive.

**Reply by abhicmraja** [reply]
The vivado-on-silicon-mac repo is on my external drive only... I have two copies actually. One on my internal drive, and one on my external drive. I make sure to run the setup.sh present on the external drive. Could the copy file on my internal drive cause an issue?

**Reply by ichi4096** [reply]
The copies shouldn't interfere with each other. For me, installing on a 1 TB external drive presented no issue, however, [it seems that Docker Desktop has an issue with reporting correct file system sizes](https://github.com/docker/for-mac/issues/7442) or maybe chooses deliberately to not report all of the free space.

The other mount options mentioned in that link may be very slow or not compatible at all with the virtualization used by Apple Silicon macs.

I can't fix that unfortunately, all I can advise is to try reinstalling Docker (and deleting the config file mentioned in the link) and redownload my repo.

**Reply by abhicmraja** [reply]
I removed the file .config file, but I'm still getting the same results.

Just to make sure, Docker can stay on the internal drive. You put the repo on the external drive, and run the .sh file, and it should work. Is that all that has to be done? Am I missing something? Should I add my volume for file sharing? (By default /Volumes is there.)

Because whether I run the setup.sh file on my internal drive or on the external drive, the "/run/host_mark/Volumes" has only 229GB and 174GB has been used.

**Reply by ichi4096** [reply]
I was just about to describe how it worked for me, then I tested it again with the same code I knew had worked fine a few weeks ago and I get the same error... Now I can at least reproduce it.

I'll try to fix it, however, my suspicion is that it is an issue related to Docker or Rosetta 2 or maybe accurate disk space readings aren't even meant to be supported.

**Reply by abhicmraja** [reply]
Ohh ok. xD

Would installing it on my internal drive first (I'll free up the space) and then moving the container to my external drive be an easy process?

**Reply by ichi4096** [reply, marked as answer]
(Edit: Didn't see your new message. Your proposed method should work but there's an easier workaround described below.)

From what I can gather, if /Volumes is shared with the container, then mounts in the Linux containers of it and all of its subfolders report the free space of / in macOS. One workaround is to remove /Volumes and add /Volumes/AbhiAPFS or wherever your external drive is mounted to in macOS to the list of shared folders in Docker Desktop -> Settings -> Resources -> File sharing.

That should work for You for the time being. I'll add a notice in the readme.md or provide a fix later (I'm thinking of double-mounting the Volume in the user's home folder so that the data in question isn't under /Volumes exclusively anymore).

**Reply by abhicmraja** [reply]
<img width="1440" alt="Screenshot 2024-10-09 at 8 53 22 PM" src="https://github.com/user-attachments/assets/3852c7af-9919-43dc-b4ae-856b0ffa3e29">

Yep, I can confirm that removing /Volumes from Shared Folders works like a charm.

Thanks a lot for your help!

**Reply by ichi4096** [reply]
You're welcome :-)

**Reply by abhicmraja** [reply]
Just wanted to follow up.... 
The installation is complete (the Vivado folder is around 47GB), but one process got killed midway, and it's asking me to run cleanup.sh and redo the installation. Is there a way to "resume" the installation/verification?

The screenshot of the error:
<img width="570" alt="Screenshot 2024-10-10 at 9 37 17 AM" src="https://github.com/user-attachments/assets/29a46fca-b993-45a0-b431-06614e16a1f6">


[Edit]: When I say installation is complete, I meant that Vivado seems to be installed. During the full installation process I got an error

**Reply by ichi4096** [reply]
No clue what that could be. There is no "resume" function that I'm aware of.

**Reply by abhicmraja** [reply]
Ohh ok. I tried installing Vivado 3 seperate times. Even redownloaded the github repo and also the Vivado web installer. But I keep getting this error.

**Reply by ichi4096** [reply]
Maybe just see if it works despite the error? Which version of Vivado are You using?

**Reply by abhicmraja** [reply]
The container doesn't get created because it exits before full completion. 
I'm using 2024.1 (the latest)

I'm getting the above error sometimes in the middle of the downloading process, and sometimes after the installation process. Is it a memory issue? Like I have allocated 4GB memory and 4GB swap. Is this enough? Or should I increase it?

**Reply by ichi4096** [reply]
Yes, try increasing both ram and swap.
